paper-api/src/main/java/io/papermc/paper/registry/data/dialog/specialty/ConfirmationSpecialty.java

```diff
@@ -0,0 +1,27 @@
+package io.papermc.paper.registry.data.dialog.specialty;
+
+import io.papermc.paper.registry.data.dialog.ActionButton;
+import org.jetbrains.annotations.Contract;
+
+/**
+ * Represents a confirmation dialog specialty.
+ * This interface defines the structure for a confirmation dialog with "confirm" and "deny" buttons.
+ */
+public sealed interface ConfirmationSpecialty extends DialogSpecialty permits ConfirmationSpecialtyImpl {
```

Comment:
The specialty naming seems a bit weird to me (also because the connotation is different to 'specific'), I would just go with ConfirmationType or ConfirmationDialogType etc., both for making it easier to find them and understanding what they are?